gguf-py/gguf/gguf.py

```diff
@@ -388,15 +413,21 @@ def get_type(val):
 
 
 class GGUFWriter:
+    fout: BinaryIO
+    arch: str
+    offset_tensor = 0
+    data_alignment = GGUF_DEFAULT_ALIGNMENT
+    kv_data = b""
+    kv_data_count = 0
+    ti_data = b""
+    ti_data_count = 0
+    use_temp_file: bool
+    temp_file: Optional[tempfile.SpooledTemporaryFile[bytes]] = None
+    tensors: List[Tuple[np.ndarray, int]]
+
     def __init__(self, path: str, arch: str, use_temp_file = True):
```

Comment:
path should be `os.PathLike[str]` so Path is accepted like on line 870 of convert.py.